Sources/MapboxMaps/Foundation/Extensions/Core/MapOptions.swift

```diff
@@ -0,0 +1,78 @@
+extension MapboxCoreMaps.MapOptions {
+    /// Initialize a `MapOptions` object that is used when initializing a Map.
+    ///
+    /// For initializing a `MapView` please see `MapInitOptions`, and
+    /// `MapOptions.default` for a convenient object that can be used in
+    /// conjunction.
+    ///
+    /// - Parameters:
+    ///   - constrainMode: The map constrain mode; default is `.heightOnly`.
+    ///   - viewportMode: The viewport mode; default is `.default`.
+    ///   - orientation: The view orientation; default is `.upwards`.
+    ///   - crossSourceCollisions: Whether cross-source symbol collision detection should be enabled; default is `true`
+    ///   - size: Size of the map, if nil (the default), a minimal default size will be used.
+    ///   - pixelRatio: Pixel scale of the map view; default is thee main screen's scale.
+    ///   - glyphsRasterizationOptions: A `GlyphsRasterizationOptions` object.
+    public convenience init(constrainMode: ConstrainMode = .heightOnly,
+                            viewportMode: ViewportMode = .default,
+                            orientation: NorthOrientation = .upwards,
+                            crossSourceCollisions: Bool = true,
+                            size: CGSize? = nil,
+                            pixelRatio: CGFloat = UIScreen.main.scale,
+                            glyphsRasterizationOptions: GlyphsRasterizationOptions = GlyphsRasterizationOptions(fontFamilies: [])) {
+
+        let mbmSize: Size?
+
+        if let size = size {
+            mbmSize = Size(width: Float(size.width), height: Float(size.height))
+        } else {
+            mbmSize = nil
+        }
+
+        self.init(__contextMode: nil,
+                  constrainMode: constrainMode.NSNumber,
+                  viewportMode: viewportMode.NSNumber,
+                  orientation: orientation.NSNumber,
+                  crossSourceCollisions: crossSourceCollisions.NSNumber,
+                  size: mbmSize,
+                  pixelRatio: Float(pixelRatio),
+                  glyphsRasterizationOptions: glyphsRasterizationOptions)
+    }
+
+    /// The map constrain mode. This can be used to limit the map to wrap around
+    /// the globe horizontally. Default is `.heightOnly`.
+    internal var constrainMode: ConstrainMode {
+        return __constrainMode?.intValueAsRawRepresentable() ?? .heightOnly
+    }
+
+    /// The viewport mode. This can be used to flip the vertical orientation of
+    /// the map as some devices may use inverted orientation.
+    internal var viewportMode: ViewportMode? {
+        return __viewportMode?.intValueAsRawRepresentable()
+    }
+
+    /// The orientation of the Map. Default is `.upwards`.
+    internal var orientation: NorthOrientation {
+        return __orientation?.intValueAsRawRepresentable() ?? .upwards
+    }
```

Comment:
Should these be public?

Reply:
Let's leave as internal for now.

Reply:
I'm okay with that. Could you share the rationale so I can follow along?

Reply:
Yep :/

Reply:
Ticketed.